desc/compute/data_index.py

```diff
@@ -75,7 +80,55 @@ def _decorator(func):
             "coordinates": coordinates,
             "dependencies": deps,
         }
-        data_index[name] = d
+        for p in parameterization:
+            flag = False
+            for base_class, superclasses in _class_inheritance.items():
+                if p in superclasses or p == base_class:
+                    data_index[base_class][name] = d.copy()
+                    flag = True
+            if not flag:
+                raise ValueError(
+                    f"Can't register function with unknown parameterization: {p}"
+                )
         return func
 
     return _decorator
+
+
+# This allows us to handle subclasses whos data_index stuff should inherit
+# from parent classes.
+# This is the least bad solution I've found, since everything else requires
+# crazy circular imports
+# could maybe make this fancier with a registry of compute-able objects?
+_class_inheritance = {
+    "desc.equilibrium.equilibrium.Equilibrium": [],
+    "desc.geometry.curve.FourierRZCurve": [
+        "desc.geometry.core.Curve",
+    ],
+    "desc.geometry.curve.FourierXYZCurve": [
+        "desc.geometry.core.Curve",
+    ],
+    "desc.geometry.curve.FourierPlanarCurve": [
+        "desc.geometry.core.Curve",
+    ],
+    "desc.geometry.surface.FourierRZToroidalSurface": [
+        "desc.geometry.core.Surface",
+    ],
+    "desc.geometry.surface.ZernikeRZToroidalSection": [
+        "desc.geometry.core.Surface",
+    ],
+    "desc.coils.FourierRZCoil": [
```

Comment:
do we also want `desc.coils.Coil` for these? For methods that would belong to Coil but not Curve like computing biot-savart

Reply:
Or we can leave for now and I can add in the coils PR #494

Reply:
Yeah currently there aren't any compute methods for the generic coil class, but if we later add some we'd need to add that here.